apps/aecore/src/aec_consensus_hc.erl

```diff
@@ -845,15 +845,17 @@ is_leader_valid(Node, _Trees, TxEnv, _PrevNode) ->
             aec_conductor:throw_error(parent_chain_block_not_synced)
     end.
 
-validate_pin(TxEnv, Trees) ->
+validate_pin(TxEnv, Trees, CurEpochInfo) ->
     case aec_chain_hc:pin_info({TxEnv, Trees}) of
         undefined -> pin_missing;
         EncTxHash ->  
             % TODO make this code much more robust - incorrect EncTxHash, bad value from PC, incorrect hash etc.etc
             lager:debug("PINNING: EncHash: ~p", [EncTxHash]),
             try 
-                #{epoch := _PinEpoch, height := PinHeight, block_hash := PinHash} = 
+                #{epoch := CurEpoch} = CurEpochInfo,
+                #{epoch := PinEpoch, height := PinHeight, block_hash := PinHash} = 
                     aec_parent_connector:get_pin_by_tx_hash(EncTxHash),
+                PinEpoch = CurEpoch - 1, % validate it was actually last epoch
```

Comment:
This will crash if it isn't true - someone complained that crashes in consesus-module (called by worker in conductor) was a bit opaque, perhaps use `aec_conductor:throw_error/1`?

Reply:
I believe this will be caught by the (very wide) `catch` on the outside, resulting in an `pin_validation_fail`?